network/src/node.rs

```diff
@@ -119,106 +143,168 @@ impl<S: Storage + Send + Sync + 'static> Node<S> {
+                        //  all of our connected peers anyways.
+
+                        // The order of preference for the sync node is as follows:
+                        //   1. Iterate (in declared order) through the bootnodes:
+                        //      a. Check if this node is connected to the specified bootnode in the peer book.
+                        //      b. Select the specified bootnode as the sync node if this node is connected to it.
+                        //   2. If this node is not connected to any bootnode,
+                        //      then select the last seen peer as the sync node.
+
+                        // Step 1.
+                        let mut sync_node = None;
+                        for bootnode in bootnodes.iter() {
+                            if node_clone.peer_book.is_connected(*bootnode) {
+                                sync_node = Some(*bootnode);
+                                break;
+                            }
+                        }
 
-                    // make sure that the node doesn't remain in a sync state without peers
-                    if self_clone.state() == State::Syncing && self_clone.peer_book.number_of_connected_peers() == 0 {
-                        self_clone.set_state(State::Idle);
-                    }
+                        // Step 2.
+                        if sync_node.is_none() {
+                            // Select last seen node as block sync node.
+                            sync_node = node_clone.peer_book.last_seen();
+                        }
 
-                    // report node's current state
-                    trace!("Node state: {:?}", self_clone.state());
+                        sync_clone.update_memory_pool(sync_node).await;
+                    }
                 }
             });
-            self.register_task(state_mgmt_task);
-
-            if let Some(ref consensus) = self.consensus() {
-                let self_clone = self.clone();
-                let consensus = Arc::clone(consensus);
-                let transaction_sync_interval = consensus.transaction_sync_interval();
-                let tx_sync_task = task::spawn(async move {
-                    loop {
-                        sleep(transaction_sync_interval).await;
-
-                        if !consensus.is_syncing_blocks() {
-                            info!("Updating transactions");
-
-                            // select last seen node as block sync node
-                            let sync_node = self_clone.peer_book.last_seen();
-                            consensus.update_transactions(sync_node).await;
+            self.register_task(sync_mempool_task);
+
+            let node_clone = self.clone();
+            let sync_clone = Arc::clone(sync);
+            let block_sync_interval = sync_clone.block_sync_interval();
+            let sync_block_task = task::spawn(async move {
+                loop {
+                    sleep(block_sync_interval).await;
```

Comment:
this delay being put here as opposed to the end of the body of the `loop` imposes an initial delay for the block sync; might be better to put it at the end instead

Reply:
Good catch, thanks!